File: @LV1/1.3/nastyhacks/nastyhacks.hs
import Control.Monad

advertise :: Ord a => a -> a -> String
advertise d c | d == c    = "does not matter"
              | d > c     = "advertise"
              | otherwise = "do not advertise"

main :: IO ()
main = do
  n <- readLn
  replicateM_ n $ do
    [r, e, c] <- map read . words <$> getLine
    putStrLn $ advertise (e - r) c

File: @LV1/1.3/nastyhacks/nastyhacks.cpp
#include <cstdio>

int main() {
    int r, e, c, n;
    scanf("%d", &n);

    for (int i = 0; i < n; i++) {
        scanf("%d %d %d", &r, &e, &c);
        int diff = e - r;

        if (diff == c) {
            puts("does not matter");
        } else if (diff > c) {
            puts("advertise");
        } else {
            puts("do not advertise");
        }
    }
}
